### PR TITLE
Update to new API

### PR DESCRIPTION
**Pull request by guyhay**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [ ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

*
*
*

### CODE REVIEW

**Comment by azure-quickstarts** [comment]
@JasonWHowell - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Review by bmoore-msft — CHANGES_REQUESTED** [review]
See the pipeline results for some cleanup that's needed

**Comment by azure-quickstarts** [comment]
@JasonWHowell - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by guyhay** [comment]
> See the pipeline results for some cleanup that's needed

Cleanup completed

File: quickstarts/microsoft.hdinsight/hdinsight-spark-linux/main.bicep [hunk 1]
```diff
@@ -52,7 +52,7 @@ param headNodeVirtualMachineSize string = 'Standard_E8_v3'
 ])
 param workerNodeVirtualMachineSize string = 'Standard_E8_v3'
 
-resource defaultStorageAccount 'Microsoft.Storage/storageAccounts@2021-04-01' = {
+resource storage_id 'Microsoft.Storage/storageAccounts@2021-04-01' = {
```
Comment (bmoore-msft) [inline quickstarts/microsoft.hdinsight/hdinsight-spark-linux/main.bicep:55]: remove ```_id``` from the symbolic name

File: quickstarts/microsoft.hdinsight/hdinsight-spark-linux/main.bicep [hunk 2]
```diff
@@ -61,7 +61,7 @@ resource defaultStorageAccount 'Microsoft.Storage/storageAccounts@2021-04-01' =
   kind: 'StorageV2'
 }
 
-resource cluster 'Microsoft.HDInsight/clusters@2018-06-01-preview' = {
+resource clusterName_resource 'Microsoft.HDInsight/clusters@2021-06-01' = {
```
Comment (bmoore-msft) [inline quickstarts/microsoft.hdinsight/hdinsight-spark-linux/main.bicep:64]: remove ```_resource``` from the name

**Review by bmoore-msft — CHANGES_REQUESTED** [review]
Still have some decompile fluff, I marked the source for each

File: quickstarts/microsoft.hdinsight/hdinsight-spark-linux/main.bicep (hunk 1 above)
Comment (guyhay) [reply quickstarts/microsoft.hdinsight/hdinsight-spark-linux/main.bicep:55]: removed

File: quickstarts/microsoft.hdinsight/hdinsight-spark-linux/main.bicep (hunk 2 above)
Comment (guyhay) [reply quickstarts/microsoft.hdinsight/hdinsight-spark-linux/main.bicep:64]: removed Name_resource

**Comment by azure-quickstarts** [comment]
@JasonWHowell - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

File: quickstarts/microsoft.hdinsight/hdinsight-spark-linux/main.bicep (hunk 1 above)
Comment (guyhay) [reply quickstarts/microsoft.hdinsight/hdinsight-spark-linux/main.bicep]: removed again

**Comment by azure-quickstarts** [comment]
@JasonWHowell - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

File: quickstarts/microsoft.hdinsight/hdinsight-spark-linux/main.bicep (hunk 2 above)
Comment (bmoore-msft) [reply quickstarts/microsoft.hdinsight/hdinsight-spark-linux/main.bicep:64]: It's back

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

**Pull request closed without merging**